### ISSUE TITLE
does this work?

### ISSUE THREAD

**Issue by jake-g** [knoopx/slsk-io#1]
thought about making my own, but came across this. Can't get it to work though...

**Comment by knoopx** [comment]
I don't think so, I did some progress but I can't remember what I got working. If you happen to work on it just tell me, maybe I join you

**Comment by artelse** [comment]
I am also interested in a slsk port to javascript / node. I tried nslsk, but ran into some problems. I fixed the bigint dependency, but then got stuck / distracted. Can you give us a head start on how to setup the dev system? Would be really helpful! Thanks!

**Comment by artelse** [comment]
Did the following:
`npm install` to install dependencies
`gulp` to compile it
It returned a few errors as:
`Error: Cannot find module 'react-immutable-proptypes' from..`
`Error: Cannot find module 'sugar' from..`
These can be fixed by:
`npm install --save react-immutable-proptypes`
`npm install --save sugar`
It compiles and is waiting, but nothing to be seen in the browser with `build/index.html`
Ok, one needs to start electron by adding to `package.json` a script directive:
`"scripts": {
    "test": "echo \"Error: no test specified\" && exit 1",
    "start": "electron ."
  },`
Now one can start by typing:
`npm start`
What next?

**Comment by artelse** [comment]
Update: I have updated most dependencies, have it running on Node 7.0.0 and Electron 1.4.5. It brings up the interface and connects to the slsk server. It complaints about an `Object.merge()` in `Column.cjsx` but cannot find any documentation on that function. I replaced it with `Object.assign()` and it works sort of, but then get into unknown props like `flex` on `<div>` tags, etc. 
@knoopx Do you have a suggestion?

**Comment by knoopx** [comment]
Object.merge comes from sugar.js but the last version does no longer monkey patch the js core library. I switched to es6 long ago and I no longer use sugar/coffee/vanilla react. The project had some missing deps in package.json and some react warnings that were previously silenced. The flex props warnings are because the layout component are forwarding the flex and some other props to the inner html div element. 

I started modernizing this project (w/ es6,  webpack and mobx) I can push what I got in a new branch if you want. 

The nslsk library that I was using is incomplete and I would like to re-implement it also using some binary  packing/unpacking library instead of rolling my own.

If you have any suggestion or deeper experience in writing binary protocol clients that would be awesome.

**Comment by artelse** [comment]
Hi, thanks for the response! I am pretty fresh on this stack, so yes, please push your current code on a new branch. Likely this branch will be deprecated anyway and we'll be working on the same code. I have experience in low level stuff as I have programmed microcontrollers and so on for a long time. So yeah, I am doing this anyway as a learning experience and because I need a good slsk client and the knowledge for a few other projects.
